### PR TITLE
Save .996s of init time by not initializing shards rods and coil in every single window, and a one line check in turf/Initialize

### PR DESCRIPTION
**Pull request by Mothblocks**
/obj/machinery/door/window/Initialize lowers by 221.81ms.

/turf/Initialize lowers by 775ms.

### CODE REVIEW

File: code/game/turfs/turf.dm [hunk 1]
```diff
@@ -118,7 +118,8 @@ GLOBAL_LIST_EMPTY(station_turfs)
 			T.multiz_turf_new(src, UP)
 
 	// by default, vis_contents is inherited from the turf that was here before
-	vis_contents.Cut()
+	if (length(vis_contents))
```
Comment (Mothblocks) [inline code/game/turfs/turf.dm:121]: This is the one line that saved like 0.7s

File: code/game/atoms_movable.dm [hunk 2]
```diff
@@ -195,7 +195,9 @@
 
 
 	vis_locs = null //clears this atom out of all viscontents
-	vis_contents.Cut()
+
+	if (length(vis_contents))
```
Comment (LemonInTheDark) [inline code/game/atoms_movable.dm:199]: document this please

File: code/game/turfs/turf.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/game/turfs/turf.dm:121]: here too then

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

**Comment by Mothblocks** [comment]
@LemonInTheDark done

**Review by LemonInTheDark — APPROVED** [review]
does changing vis_locs affect anything? or no

**Comment by Mothblocks** [comment]
changing it where

**Comment by LemonInTheDark** [comment]
image

**Comment by Mothblocks** [comment]
Uhh I can look if you want it for this PR but I have a bunch of other stuff I'm doing

**Comment by LemonInTheDark** [comment]
sall good

**Pull request merged**